### DISCUSSION TITLE
T-CNC carrier board

### DISCUSSION THREAD

**Discussion by philGrz** [grblHAL/Controllers#4] [General]
i am currently working on the hardware for a Teensy 4.1 CNC controller

see actual status on 

https://github.com/philGrz/Controllers/wiki/T-CNC

**Comment by terjeio** [comment]
Any progress with this work?

**Comment by philGrz** [comment]
Hi,

 

summer and holidays are over 😊

 

i will continue soon.

 

Can you do the port of GRBL32 ?

 

BR

 

Philipp

 

Von: Terje Io ***@***.***> 
Gesendet: Dienstag, 27. September 2022 07:49
An: grblHAL/Controllers ***@***.***>
Cc: Phil ***@***.***>; Author ***@***.***>
Betreff: Re: [grblHAL/Controllers] T-CNC carrier board (Discussion #4)

 

Any progress with this work?

**Reply by terjeio** [reply]
> Can you do the port of GRBL32 ?

Which one (link please)? And why would I want to do that? Are you missing some specific functionality in grblHAL?

**Comment by philGrz** [comment]
Hi,

 

no thats not the thing.

 

I am hardware engineer and only do some minor software things, my partner who was in charge to do the SW and additional code to operate the IOs cant do that currently due to family reasons.

 

T CNC · philGrz/Controllers Wiki (github.com) <https://github.com/philGrz/Controllers/wiki/T-CNC> 

 

Required functions:

2 pins to independently drive the double X for gantry squaring

4 I2C chips with 3 interrupts (limit switches, fault inputs, digital inputs) for IO 

 

Optional functions:

1 I2C OLED

RS485 propably Modbus

 

The hardware parts are available but not finally assembled

 

 

BR

 

Philipp

 

 

Von: Terje Io ***@***.***> 
Gesendet: Dienstag, 27. September 2022 13:09
An: grblHAL/Controllers ***@***.***>
Cc: Phil ***@***.***>; Author ***@***.***>
Betreff: Re: [grblHAL/Controllers] T-CNC carrier board (Discussion #4)

 

Can you do the port of GRBL32 ?

Which one (link please)? And why would I want to do that? Are you missing some specific functionality in grblHAL?

**Reply by terjeio** [reply]
>  2 pins to independently drive the double X for gantry squaring

Already supported.

> 4 I2C chips with 3 interrupts (limit switches, fault inputs, digital inputs) for IO

Why on earth do you want that? The Teensy 4.1 should have plenty pins for direct inputs for interrupt capable pins. Better to add an ioexpander for slow output signals if lack of pins. However, can be done via board specific code if you insist so.

> 1 I2C OLED

I am not going to add displays directly connected to the controller, you may do so yourself via a plugin. I2C support is in place.

> RS485 propably Modbus

Already supported but might need code for an additional UART port. Easy to add.
